src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -506,9 +506,11 @@ internal static unsafe IntPtr ComInterfaceForObject(object instance, Guid target
             int resultCode = wrapper->QueryInterface(in targetIID, out comObjectInterface);
             if (resultCode != 0)
             {
-                throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+                throw new PlatformNotSupportedException(SR.InvalidOperation_InterfaceNotSupported);
```

Comment:
Is this the exception that CoreCLR throws in this situation? (I would expect it to be something else.)

Reply:
Cannot find location. Will try again when would be home in hour or so.

Reply:
Is this proper location where look for object -> interface conversion.

https://github.com/dotnet/runtime/blob/aaaaeed8c8d2d01dccd1f7fc536ea27eee75f472/src/coreclr/vm/ilmarshalers.cpp#L1401

Maybe Marshal class will throw this exception too.
I account for case where ComWrappers does not provide correct interface vtbl

Reply:
The easiest way to find out is to write a small test.